daemon/daemon.go

```diff
@@ -726,23 +712,33 @@ func NewDaemon(config *Config, registryService *registry.Service) (daemon *Daemo
 	logrus.Debugf("Using default logging driver %s", config.LogConfig.Type)
 
 	// Configure and validate the kernels security support
-	if err := configureKernelSecuritySupport(config, d.driver.String()); err != nil {
+	if err := configureKernelSecuritySupport(config, config.GraphDriver); err != nil {
 		return nil, err
 	}
 
 	daemonRepo := filepath.Join(config.Root, "containers")
-
 	if err := idtools.MkdirAllAs(daemonRepo, 0700, rootUID, rootGID); err != nil && !os.IsExist(err) {
 		return nil, err
 	}
 
-	imageRoot := filepath.Join(config.Root, "image", d.driver.String())
+	imageRoot := filepath.Join(config.Root, "image", config.GraphDriver)
```

Comment:
Oh ya, this won't work. If the user doesn't specify a graph driver the path is going to be wrong.  This needs to happen after `graphdriver.New()` does the work to determine the existing driver.